src/biz/tugay/groovyship/service/BoardService.groovy

```diff
@@ -0,0 +1,56 @@
+package biz.tugay.groovyship.service
+
+import biz.tugay.groovyship.modal.Board
+import biz.tugay.groovyship.modal.Ship
+import biz.tugay.groovyship.modal.Coordinate
+
+class BoardService
+{
+
+  ShipService shipService = new ShipService()
+
+  /**
+   * Attempts to add the {@code ship} to the board. As per the rules of the game, a ship cannot overlay with
+   * any other existing ship on the board. Ships also cannot be adjacent to each other.
+   *
+   * @param board The board the ship will be added to, potentially having ships already added.
+   * @param ship The new ship that is being attempted to be added.
+   * @return Whether ship was added to the board or not.
+   */
+  boolean addShip(Board board, Ship ship) {
+    def shipCoordinates = ship.coordinateIsHitByMissileMap.keySet()
+
+    if (board.ships.any { shipService.isBoundariesOfShipInCoordinates it, shipCoordinates }) {
+      return false
+    }
+
+    board.ships.add(ship)
+    return true
+  }
+
+  /**
+   * Sends a missile to to the board. Returns whether it hit a ship or not.
+   *
+   * @param board The board the missile be sent to
+   * @param column The column of the missile
+   * @param row The row of the missle
+   * @return Whether a ship was hit or not
+   */
+  boolean missileCoordinate(Board board, int column, int row) {
+    def missileCoordinate = Coordinate.of column, row
+    board.missileAttempts << missileCoordinate
```

Comment:
`<<` is equivalent of `add`.

Reply:
i saw similar just today in some post, i thought it was about bitwise operators. the following is interesting:
`sql_sel.where(tbl_cols.tg_chat_id == chat_id, tbl_cols.tg_user_id == tg_user_id)`
 tbl_cols is actually similar to a map but its keys are appearing as properties. Also expressions are passed to the method. Related to eq function implementation this is possible and does not evaluate the expression